src/solver/hierarchical-iterative.cc

```diff
@@ -830,6 +831,8 @@ namespace hpp {
             else
               d.PK.noalias() = *projector * getV2<SVD_t> (d.svd, rank);
             projector = &d.PK;
+            if(solveLevelByLevel_ && err.squaredNorm() > squaredErrorThreshold_)
+              break;
```

Comment:
You can do this before computing the projector for the next iteration.

Reply:
Like this ?

Reply:
Yes.